js/plugins/iq.js

```diff
@@ -257,22 +257,29 @@ var IqPlugin= function (framework)
       case OO.Analytics.EVENTS.AD_COMPLETED:
       case OO.Analytics.EVENTS.AD_CLICKTHROUGH_OPENED:
       case OO.Analytics.EVENTS.SDK_AD_EVENT:
-        var eventMetadata = params[0];
-        if(!eventMetadata)
+        if (this.ooyalaReporter)
         {
-          eventMetadata = {};
-        }
+          var eventMetadata = params[0];
+          if(!eventMetadata)
+          {
+            eventMetadata = {};
+          }
 
-        if (eventMetadata.adEventName)
-        {
-          eventMetadata.adEventName = eventName + ":" + eventMetadata.adEventName;
+          if (eventMetadata.adEventName)
+          {
+            eventMetadata.adEventName = eventName + ":" + eventMetadata.adEventName;
+          }
+          else
+          {
+            eventMetadata.adEventName = eventName;
+          }
+          this.ooyalaReporter.reportCustomEvent(eventName, eventMetadata);
+          OO.log("IQ: Reported: reportCustomEvent() for event: " + eventName + " with args:" + JSON.stringify(eventMetadata));
         }
         else
         {
-          eventMetadata.adEventName = eventName;
+          OO.log("IQ: Tried reporting event: " + eventName + " but ooyalaReporter is: " + this.ooyalaReporter);
```

Comment:
I think OO.log or console.log doesn't like null values(this.ooyalaReporter).  Can you make sure you test this path and see if it actually logs this error.

Reply:
I tested it, it prints out ".. but ooyalaReporter is: null" in the console

Reply:
yay!